src/middlewared/middlewared/plugins/vm/disk_utils.py

```diff
@@ -40,7 +40,10 @@ def import_disk_image(self, job, data):
         `diskimg` is an required parameter for the incoming disk image
         `zvol` is the required target for the imported disk image
         """
-        if not self.middleware.call_sync('zfs.dataset.query', [('id', '=', data['zvol'])]):
+        if not self.middleware.call_sync(
```

Comment:
I think we should add changes so that we do not allow this to happen if source is in a system managed dataset. Thoughts ?